text/2024-01-27-autohinting.md

```diff
@@ -0,0 +1,266 @@
+# Autohinting
+
+*Or, how to smash a glyph into the pixel grid without any help from the font
+designer.*
+
+[Hinting](https://en.wikipedia.org/wiki/Font_hinting) is the process of
+generating information (aka "hints") for a font that define how glyph outlines
+should be adjusted at various sizes so that the edges better align with the
+pixel grid when rasterized. This improves contrast and readability at small sizes. 
+ 
+Manual hinting is done explicitly by the font developer, typically through the
+use of graphical tools or writing TrueType instructions directly.
+
+*Auto*hinting is done by software, using a set of heuristics to determine how
+outlines should be adjusted. This can be done ahead of time
+(see [ttfautohint](https://freetype.org/ttfautohint/)) or just in time as glyph
+outlines are requested from the font. This document will focus on the latter.
```

Comment:
Suggest noting Android makes heavy use of autohinting and linking to the source. Potentially also not ethis is arguably unnecessary for a modern hires screen.